### PR TITLE
Implement open_orders and open_orders_by_trader_id (with tests)

### PR DESCRIPTION
**Pull request by teixeiramarcia**

### CODE REVIEW

File: apps/exchange/lib/exchange/matching_engine.ex [hunk 1]
```diff
@@ -201,6 +217,20 @@ defmodule Exchange.MatchingEngine do
...
   end
 
+  def handle_call(:open_orders, _from, order_book) do
+    open_orders =
+      Exchange.OrderBook.open_orders(order_book)
+
+    {:reply, {:ok, open_orders}, order_book}
+  end
+
+  def handle_call(:open_orders_by_trader, trader_id, order_book) do
```
Comment (lmpn) [inline apps/exchange/lib/exchange/matching_engine.ex:227]: It should be
`def handle_call({:open_orders_by_trader, trader_id}, _from, order_book) do`

File: apps/exchange/lib/exchange/matching_engine.ex [hunk 2]
```diff
@@ -114,6 +114,22 @@ defmodule Exchange.MatchingEngine do
...
+  @spec open_orders(ticker) :: {atom, list()}
+  def open_orders(ticker) do
+    GenServer.call(via_tuple(ticker), :open_orders)
+  end
+
+  @doc """
+  Returns the list of open orders from a trader
+  """
+  @spec open_orders_by_trader(ticker) :: {atom, list()}
+  def open_orders_by_trader(ticker) do
```
Comment (lmpn) [inline apps/exchange/lib/exchange/matching_engine.ex:129]: Here you need to receive as input the trader_id.
`@spec open_orders_by_trader(ticker, string.t()) :: {atom, list()}`
`def open_orders_by_trader(ticker, trader_id) do`

File: apps/exchange/lib/exchange.ex [hunk 3]
```diff
@@ -93,6 +93,10 @@ defmodule Exchange do
     Exchange.MatchingEngine.ask_volume(ticker)
   end
 
+  def open_orders(ticker) do
+    Exchange.MatchingEngine.open_orders(ticker)
+  end
+
```
Comment (lmpn) [inline apps/exchange/lib/exchange.ex:99]: Here you should have a call for the open_orders_trader_id.
The arguments are the ticker and trader_id

File: apps/exchange/lib/exchange/order_book.ex [hunk 4]
```diff
@@ -535,4 +535,32 @@ defmodule Exchange.OrderBook do
...
+    open_sell_orders = orders_to_list(order_book.sell)
+    open_buy_orders = orders_to_list(order_book.buy)
+
+    open_sell_orders ++ open_buy_orders
+  end
+
+  @spec orders_to_list(Map.t()) :: list()
+  def orders_to_list(orders) do
+    orders
+      |> Map.to_list()
```
Comment (lmpn) [inline apps/exchange/lib/exchange/order_book.ex:553]: This line is really necessary? 🤔

File: apps/exchange/lib/exchange/matching_engine.ex (hunk 2 above)
Comment (teixeiramarcia) [reply apps/exchange/lib/exchange/matching_engine.ex:129]: Done

**Pull request merged**